artifactory/utils/buildutils.go

```diff
@@ -265,18 +270,119 @@ func RemoveBuildDir(buildName, buildNumber, projectKey string) error {
 }
 
 type BuildConfiguration struct {
-	BuildName   string
-	BuildNumber string
-	Module      string
-	Project     string
+	buildName            string
+	buildNumber          string
+	module               string
+	project              string
+	loadedFromConfigFile bool
+}
+
+func NewBuildConfiguration(buildName, buildNumber, module, project string) *BuildConfiguration {
+	return &BuildConfiguration{buildName: buildName, buildNumber: buildNumber, module: module, project: project}
+}
+
+func (bc *BuildConfiguration) SetBuildName(buildName string) *BuildConfiguration {
+	bc.buildName = buildName
+	return bc
+}
+
+func (bc *BuildConfiguration) SetBuildNumber(buildNumber string) *BuildConfiguration {
+	bc.buildNumber = buildNumber
+	return bc
+}
+
+func (bc *BuildConfiguration) SetProject(project string) *BuildConfiguration {
+	bc.project = project
+	return bc
+}
+
+func (bc *BuildConfiguration) SetModule(module string) *BuildConfiguration {
+	bc.module = module
+	return bc
+}
+
+func (bc *BuildConfiguration) GetBuildName() (string, error) {
+	if bc.buildName != "" {
+		return bc.buildName, nil
+	}
+	// Resolve from env var.
+	if envValue := os.Getenv(coreutils.BuildName); envValue != "" {
+		bc.buildName = envValue
+		return bc.buildName, nil
+	}
+	// Resolve from config file in '.jfrog' folder.
+	confFilePath, exist, err := GetProjectConfFilePath(Build)
+	if err != nil || !exist {
+		return "", err
+	}
+	vConfig, err := ReadConfigFile(confFilePath, YAML)
+	if err != nil || vConfig == nil {
+		return "", err
+	}
+	if bc.buildName = vConfig.GetString(ProjectConfigBuildNameKey); bc.buildName != "" {
+		bc.loadedFromConfigFile = true
+	}
+	return bc.buildName, nil
+}
+
+func (bc *BuildConfiguration) GetBuildNumber() string {
+	if bc.buildNumber != "" {
+		return bc.buildNumber
+	}
+	// Resolve from env var.
+	if bc.buildNumber = os.Getenv(coreutils.BuildNumber); bc.buildNumber != "" {
+		return bc.buildNumber
+	}
+	// If build name was resolve from build.yaml file, use 'LATEST' as build number.
+	if buildName, err := bc.GetBuildName(); err == nil && buildName != "" && bc.loadedFromConfigFile {
+		bc.buildNumber = artclientutils.LatestBuildNumberKey
+	}
+	return bc.buildNumber
+}
+
+func (bc *BuildConfiguration) GetProject() string {
+	return bc.project
+}
+
+func (bc *BuildConfiguration) GetModule() string {
+	return bc.module
 }
 
-func ValidateBuildAndModuleParams(buildConfig *BuildConfiguration) error {
-	if (buildConfig.BuildName == "" && buildConfig.BuildNumber != "") || (buildConfig.BuildName != "" && buildConfig.BuildNumber == "") {
-		return errors.New("the build-name and build-number options cannot be provided separately")
+// Return nil if build name & number & module are found or none of them.
```

Comment:
This description isn't clear.
The same goes for the description of the ```ValidateBuildParams``` mwthod.